src/System.Net.HttpListener/tests/WebSocketTests.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Net.Http;
+using System.Net.WebSockets;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.Net.Tests
+{
+    public class WebSocketTests
+    {
+        [ConditionalFact(nameof(PlatformDetection) + "." + nameof(PlatformDetection.IsNotOneCoreUAP))]
+        public static async Task AcceptWebSocketAsync_NullSubProtocol_Succeeds()
+        {
+            string url = UrlPrefix.CreateLocal();
+
+            using (HttpListener listener = new HttpListener())
+            {
+                listener.Prefixes.Add(url);
+                listener.Start();
+
+                UriBuilder uriBuilder = new UriBuilder(url);
+                uriBuilder.Scheme = "ws";
+
+                var serverContextTask = listener.GetContextAsync();
+
+                using (ClientWebSocket clientWebSocket = new ClientWebSocket())
+                {
+                    Task clientConnectTask = clientWebSocket.ConnectAsync(uriBuilder.Uri, CancellationToken.None);
+
+                    Assert.Equal(WebSocketState.Connecting, clientWebSocket.State);
+
+                    HttpListenerContext listenerContext = await serverContextTask;
+                    HttpListenerWebSocketContext wsContext = await listenerContext.AcceptWebSocketAsync(null);
+
+                    await clientConnectTask;
+
+                    // Ensure websocket is connected from server.
+                    Assert.Equal(WebSocketState.Open, wsContext.WebSocket.State);
+
+                    // Websocket subProtocol.
+                    Assert.Null(wsContext.WebSocket.SubProtocol);
+
+                    const string expected = "hello";
+                    byte[] receiveBuffer = Encoding.UTF8.GetBytes(expected);
+
+                    // Send binary data from server.
+                    await wsContext.WebSocket.SendAsync(new ArraySegment<byte>(receiveBuffer), WebSocketMessageType.Binary, true, CancellationToken.None);
+
+                    // Receive binary data in client.
+                    ArraySegment<byte> getBuffer = new ArraySegment<byte>(new byte[receiveBuffer.Length]);
+                    await clientWebSocket.ReceiveAsync(getBuffer, CancellationToken.None);
```

Comment:
Technically the client needs to loop, as it may end up reading part of the message in one call and need one or more subsequent calls to read the rest.  That's unlikely to happen with such a short message, but it's possible.